src/core/Tools.h

```diff
@@ -101,6 +101,20 @@ namespace Tools
 
         return version;
     }
+
+    /**
+     * Securely wipe a memory buffer.
+     *
+     * @param ptr pointer to first buffer element
+     * @param size number of bytes to erase
+     */
+    inline void wipeBuffer(void* ptr, std::size_t size)
```

Comment:
I wouldn't be surprised if compilers ignore `inline` (it's only a hint) and remove calls to `wipeBuffer` function for optimization.
Clearing memory is hard, so don't roll your own...

FYI: https://media.ccc.de/v/35c3-9788-memsad

Reply:
Wow great video. We could use sodium_memzero

Reply:
inline is only a hint, yes. The volatile approach is something that's supposed to work pretty reliably. There are other ways which are slightly faster, but less reliable.

Reply:
`volatile` kind of prevents the compiler to optimize out the loop itself.
I'm talking about the call to wipeBuffer if the compiler ignores `inline` and emits a function call.

Reply:
The compiler shouldn't be allowed to optimize out a function call if it can't prove that it's free of side-effects. I also haven't seen it do it.
I have put quite a bit of research into this part, but I'd be fine with using existing memory deletion functions as well if we do not introduce new dependencies for it.